strangelove-ventures/sommelier
# Issue 305: Detect restricted countries via API route

Fixes #305

## Description

Adds detection for restricted countries. The sanctioned list was pulled from https://orpa.princeton.edu/export-controls/sanctioned-countries.

Ukraine region iso_codes were cross referenced from https://en.wikipedia.org/wiki/ISO_3166-2:UA and verified from past experience.

## Changes

- Adds an api route `GET /api/geo` that returns:
```js
{
  isRestricted: boolean,
  country: string,
  region: string,
}
```
- Adds script to download the `maxmind` database. This requires acquiring a free license key from [maxmind.com](https://maxmind.com) and setting the env var `MAXMIND_KEY`.

## Testing Steps

We rely on the `x-forwarded-for` header to figure out the client's IP address (provided by Vercel). In your local env, you wont have this header. So either use curl to set it and provide an IP address or hardcode it in `src/pages/api/geo.ts`.

```bash
curl -H 'x-forwarded-for: 175.45.176.2' localhost:3000/api/geo
```

Make sure to follow the steps in the readme to get the Maxmind City db.

## Before merging
- Update the build command to `yarn build:maxmind` in Vercel
- Configure the `MAXMIND_KEY` environment variable in Vercel
